File: scripts/convert_ordering.py
#!/usr/bin/env python3
import re
import json

def convert_ordering_questions():
    with open('client/src/data/questions.ts', 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Pattern to match ordering questions
    pattern = r'({\s*id:\s*"ordering-\d+",\s*points:\s*\d+,\s*q:\s*"[^"]*",\s*a:\s*)"([^"]+)"'
    
    def replacer(match):
        prefix = match.group(1)
        answer_str = match.group(2)
        
        # Split by ➔ to get individual items
        items = [item.strip() for item in answer_str.split('➔')]
        
        # Create JSON arrays
        items_json = json.dumps(items, ensure_ascii=False)
        correct_order_json = json.dumps(items, ensure_ascii=False)
        
        # Return with added orderItems and correctOrder
        return f'{prefix}"{answer_str}", orderItems: {items_json}, correctOrder: {correct_order_json}'
    
    # Replace all ordering questions
    converted = re.sub(pattern, replacer, content)
    
    # Count replacements
    count = len(re.findall(pattern, content))
    
    with open('client/src/data/questions.ts', 'w', encoding='utf-8') as f:
        f.write(converted)
    
    print(f'Converted {count} ordering questions')

if __name__ == '__main__':
    convert_ordering_questions()

File: scripts/convert_ordering.js
// Convert ordering questions from "item1 ➔ item2 ➔ ..." to orderItems/correctOrder format
const fs = require('fs');
const path = require('path');

// Read the data file
const dataPath = path.join(__dirname, '../client/src/data/questions.ts');
const content = fs.readFileSync(dataPath, 'utf-8');

// Find and parse ordering questions
let result = content;
let found = 0;

// Simple replacement function
function convertQuestion(line) {
  // Match lines like: { id: "ordering-1", points: 200, q: "...", a: "item1 ➔ item2 ➔ ..." }
  const match = line.match(/(\{\s*id:\s*"ordering-\d+",\s*points:\s*\d+,\s*q:\s*"[^"]*",\s*a:\s*)"([^"]+)"/);
  if (!match) return line;
  
  const prefix = match[1];
  const answer = match[2];
  
  // Check if answer contains ➔
  if (!answer.includes('➔')) {
    return line;
  }
  
  // Split by ➔ and trim
  const items = answer.split('➔').map(item => item.trim());
  
  if (items.length < 2) {
    return line;
  }
  
  // Generate the new format
  const orderItems = JSON.stringify(items);
  const correctOrder = JSON.stringify(items);
  
  // Replace in original line
  const newLine = line.replace(
    /(\{\s*id:\s*"ordering-\d+",\s*points:\s*\d+,\s*q:\s*"[^"]*",\s*a:\s*)"[^"]+"(\s*[},])/,
    `$1"${answer}", orderItems: ${orderItems}, correctOrder: ${correctOrder}$2`
  );
  
  return newLine;
}

// Process line by line
const lines = result.split('\n');
const converted = [];

for (let i = 0; i < lines.length; i++) {
  let line = lines[i];
  if (line.includes('ordering-') && line.includes('➔')) {
    const newLine = convertQuestion(line);
    if (newLine !== line) {
      found++;
    }
    converted.push(newLine);
  } else {
    converted.push(line);
  }
}

console.log(`Found and converted ${found} ordering questions`);

// Write back
fs.writeFileSync(dataPath, converted.join('\n'));
console.log('Conversion complete');
